### ISSUE TITLE
[HOLD for payment 2025-02-04] [HOLD for payment 2025-01-24] [$250] Editing a comment: the cursor is jumping to the end of the message when we type in the middle of the text making it impossible to edit the text

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#54335]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** Needs Reproduction
**Reproducible in production?:** Needs Reproduction
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @mountiny @quinthar 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1734531561566439)

## Action Performed:

1. Write a message and send it
2. Go edit the message
3. Move the cursor to the middle of the text
4. Start to type

## Expected Result:
User able to type where the cursor 

## Actual Result:
 The cursor jumps to the end of the text after a first character

## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

https://github.com/user-attachments/assets/00222bda-a28f-4572-bf42-2f6c9b11cffe

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021869733355861718010</li>
        <li>Upwork Job ID: 1869733355861718010</li>
        <li>Last Price Increase: 2024-12-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @stephanieelliott</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by allroundexperts** [comment]
Seems related to the grammar extension that they're using.

**Comment by mountiny** [comment]
I do have Grammarly installed

**Comment by allroundexperts** [comment]
Can you try without that?

**Comment by mountiny** [comment]
I cannot reproduce it now even with it on, so I will keep monitoring if it happens again

**Comment by m-natarajan** [comment]
@mountiny @allroundexperts Able to reproduce this issue 

https://github.com/user-attachments/assets/7df66eb5-668e-4100-83ef-027e5ec1c1b6

**Comment by bernhardoj** [comment]
I think this is the same as this [one](https://github.com/Expensify/App/issues/53797)

**Comment by mountiny** [comment]
Thanks! SWM will have a look in the mean time 🙌

**Comment by tomekzaw** [comment]
Someone from SWM will take a look at this shortly cc @BartoszGrajdek @Skalakid

**Comment by allroundexperts** [comment]
I can take this as I have context on it already!

**Comment by tomekzaw** [comment]
Cannot reproduce this both on new.expensify.com and staging.new.expensify.com, both on Chrome and Safari on macOS.

**Comment by Skalakid** [comment]
Are there any additional steps that we need to do to reproduce that?  On which browser does it occur?

**Comment by bernhardoj** [comment]
[This](https://github.com/Expensify/App/issues/53797) is the same issue with a better repro step.

**Comment by mountiny** [comment]
@bernhardoj can you move the proposal over here? Seems like it has stalled in that issue

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The cursor is at the beginning of the message if we open the admins room from the workspace profile page.

### What is the root cause of that problem?
The root cause of the issue is similar to https://github.com/Expensify/App/issues/46095. This is caused by the active focus traps that prevents the focus event to be received by the composer. You can read more detail on that root cause [here](https://github.com/Expensify/App/issues/46095#issuecomment-2248239994).

The different thing in this issue is, we already pause the focus trap if the screen isn't focused.
https://github.com/Expensify/App/blob/8981081bed8fb8a07f8797d49af2797332c9259d/src/components/FocusTrap/FocusTrapForScreen/index.web.tsx#L41

In this case, the screen that has the active focus trap is the workspace full screen navigator. When we open the workspace profile share page, the workspace full screen navigator focus trap is paused. When we click the #admins room link, the RHP is closed and focus trap automatically unpause the previous focus trap, which is the workspace full screen navigator focus trap.
https://github.com/focus-trap/focus-trap/blob/f85c73ac36553d952a8e03d1033b35544247caa0/index.js#L28-L36
https://github.com/focus-trap/focus-trap/blob/master/CHANGELOG.md#400

### What changes do you think we should make in order to solve the problem?

We don't want the workspace full screen navigator focus trap to be unpaused when the RHP is closed because the screen is still blurred. We have 2 options.

First, instead of pausing the focus trap when the screen is blurred, we can deactivate it. We can return `isFocused` here (or we can do it only for FullScreenNavigator to be very safe).
https://github.com/Expensify/App/blob/8981081bed8fb8a07f8797d49af2797332c9259d/src/components/FocusTrap/FocusTrapForScreen/index.web.tsx#L35-L36

If the focus trap is deactivated, unpause doesn't do anything.
https://github.com/focus-trap/focus-trap/blob/f85c73ac36553d952a8e03d1033b35544247caa0/index.js#L1037-L1040


Second, we can add a new option to the `focus-trap` repo to not automatically pause/unpause the focus-trap when a focus-trap is activated/deactivated.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


N/A

**Comment by allroundexperts** [comment]
Yep, thank you. @mountiny can you assign this to me?

**Comment by mountiny** [comment]
How does the @bernhardoj proposal look like! @allroundexperts @tomekzaw @Skalakid

**Comment by allroundexperts** [comment]
@bernhardoj's proposal looks good to me. I prefer going via the second route, but I am not sure if we want to wait too much.

**Comment by mountiny** [comment]
Thanks @bernhardoj Can you please work on the implementation? 

The `focus-trap` repository does not look like a repo that would frequently handle PRs from the community. I agree it would be nice to have this fixed upstream. Can you please look into this as a follow up (first lets fix this in App) and then try to propose your fix in the upstream repo and hope it would get accepted?

Thanks!

**Comment by bernhardoj** [comment]
PR is ready

cc: @allroundexperts 

I'll open the upstream PR later and link it on the App PR.

**Comment by stephanieelliott** [comment]
Hey @bernhardoj cross posting @allroundexperts [comment from the PR](https://github.com/Expensify/App/pull/54593#issuecomment-2567160382) -- can you give an ETA on when you'll be able to address this? This is a critical bug, so we are trying to get it unblocked ASAP: 

> @bernhardoj Can you maybe try to merge main and then we can trigger a rebuild? It's not working for me on Android web.

**Comment by stephanieelliott** [comment]
Hey @bernhardoj bump on the above ^^

**Comment by bernhardoj** [comment]
Sorry, I was OOO, addressed the comment.

**Comment by stephanieelliott** [comment]
Just noting a second PR was opened and is under review: https://github.com/Expensify/App/pull/55285

**Comment by stephanieelliott** [comment]
Summarizing payment on this issue: 

 - Contributor: @allroundexperts $250 via ND -- please request! 
 - Contributor+: @bernhardoj $250 via ND -- please request!

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by allroundexperts** [comment]
# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [x] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [x] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: As mentioned in the proposal, this was due to an issue with the focus trap library which is externally managed.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


- Log in to an account that has a workspace

### Test:


1. Open the workspace profile page
2. Press Share button
3. Press the #admins text link. This will open the admins room
4. Focus on the composer
5. Verify the composer has a green outline
6. Send any message (if there is no message yet)
7. Right-click/long-press the message and choose Edit comment

Verify the edit composer has a green outline

8. Moves the cursor to the start of the text
9. Type anything

Verify the cursor doesn't go to the end of the text

Do we agree 👍 or 👎

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
